data-plane/receiver/src/test/java/dev/knative/eventing/kafka/broker/receiver/MockReactiveProducerFactory.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright © 2018 Knative Authors (knative-dev@googlegroups.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.knative.eventing.kafka.broker.receiver;
+
+import java.util.Properties;
+
+import org.apache.kafka.clients.producer.Producer;
+
+import io.cloudevents.CloudEvent;
+import io.vertx.core.Vertx;
+
+public class MockReactiveProducerFactory implements ReactiveProducerFactory<String, CloudEvent> {
+
+    @Override
+    public ReactiveKafkaProducer<String, CloudEvent> create(Vertx v, Properties config) {
+        return new MockReactiveKafkaProducer<String, CloudEvent>();
+    }
+
+    public static ReactiveKafkaProducer<String, CloudEvent> createStatic(Vertx v,
+            Producer<String, CloudEvent> producer) {
+        return new MockReactiveKafkaProducer<String, CloudEvent>();
+    }
```

Comment:
The issue of `ReceiverVerticleTest` failing is due to not using the `producer` parameter

I did this:

`MockReactiveProducerFactory`
```java
package dev.knative.eventing.kafka.broker.receiver;

import java.util.Properties;

import org.apache.kafka.clients.producer.MockProducer;
import org.apache.kafka.clients.producer.Producer;

import io.cloudevents.CloudEvent;
import io.vertx.core.Vertx;

public class MockReactiveProducerFactory implements ReactiveProducerFactory<String, CloudEvent> {

  @Override
  public ReactiveKafkaProducer<String, CloudEvent> create(Vertx v, Properties config) {
    return new MockReactiveKafkaProducer<String, CloudEvent>(new MockProducer<>());
  }

  public static ReactiveKafkaProducer<String, CloudEvent> createStatic(Vertx v,
                                                                       Producer<String, CloudEvent> producer) {
    return new MockReactiveKafkaProducer<String, CloudEvent>(producer);
  }

}
```
`MockReactiveKafkaProducer`
```java
package dev.knative.eventing.kafka.broker.receiver;

import io.vertx.core.Promise;
import org.apache.kafka.clients.producer.Callback;
import org.apache.kafka.clients.producer.Producer;
import org.apache.kafka.clients.producer.ProducerRecord;
import org.apache.kafka.clients.producer.RecordMetadata;

import io.cloudevents.CloudEvent;
import io.vertx.core.Future;

public class MockReactiveKafkaProducer<K, V> implements ReactiveKafkaProducer<String, CloudEvent> {

  private final Producer<String, CloudEvent> producer;

  public MockReactiveKafkaProducer(Producer<String, CloudEvent> producer) {
    this.producer = producer;
  }

  @Override
  public Future<Void> close() {
    producer.close();
    return Future.succeededFuture();
  }

  @Override
  public Future<Void> flush() {
    producer.flush();
    return Future.succeededFuture();
  }

  @Override
  public Future<RecordMetadata> send(ProducerRecord<String, CloudEvent> record) {
    final Promise<RecordMetadata> p = Promise.promise();
    producer.send(record, (metadata, exception) -> {
      if (exception != null) {
        p.fail(exception);
      } else {
        p.complete(metadata);
      }
    });
    return p.future();
  }

  @Override
  public org.apache.kafka.clients.producer.Producer<String, CloudEvent> unwrap() {
    return producer;
  }
}
```